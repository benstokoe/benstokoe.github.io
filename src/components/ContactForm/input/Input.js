import React from 'react';
import classnames from 'classnames';
import './input.scss';

const Input = ({ type, error, ...props }) => {
  const Type = type === 'textarea' ? 'textarea' : 'input';

  return (
    <div className="input__container">
      <Type type="text" className={classnames('input', { 'input--error': error })} {...props} />
      {error && <span className="input__error">{error}</span>}
    </div>
  );
};

export default Input;
