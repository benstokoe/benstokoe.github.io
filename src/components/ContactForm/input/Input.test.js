import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input', () => {
  let wrapper;
  const props = {
    error: false,
    disabled: false,
  };

  beforeEach(() => {
    wrapper = shallow(<Input {...props} />);
  });

  it('should render properly with no props', () => {
    wrapper = shallow(<Input />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly for textarea', () => {
    wrapper.setProps({
      type: 'textarea',
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly for input', () => {
    wrapper.setProps({
      type: 'input',
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly with error', () => {
    wrapper.setProps({
      type: 'input',
      error: true,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
