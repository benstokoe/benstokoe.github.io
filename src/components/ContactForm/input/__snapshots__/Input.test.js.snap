// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render properly for input 1`] = `
<div
  className="input__container"
>
  <input
    className="input"
    disabled={false}
    type="text"
  />
</div>
`;

exports[`Input should render properly for textarea 1`] = `
<div
  className="input__container"
>
  <textarea
    className="input"
    disabled={false}
    type="text"
  />
</div>
`;

exports[`Input should render properly with error 1`] = `
<div
  className="input__container"
>
  <input
    className="input input--error"
    disabled={false}
    type="text"
  />
  <span
    className="input__error"
  >
    <Component />
  </span>
</div>
`;

exports[`Input should render properly with no props 1`] = `
<div
  className="input__container"
>
  <input
    className="input"
    type="text"
  />
</div>
`;
